eventmesh-connectors/eventmesh-connector-prometheus/src/main/java/org/apache/eventmesh/connector/prometheus/source/connector/PrometheusSourceConnector.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    @Override
+    public void init(Config config) throws Exception {
+        this.sourceConfig = (PrometheusSourceConfig) config;
+        doInit();
+    }
+
+    @Override
+    public void init(ConnectorContext connectorContext) throws Exception {
+        SourceConnectorContext sourceConnectorContext = (SourceConnectorContext) connectorContext;
+        this.sourceConfig = (PrometheusSourceConfig) sourceConnectorContext.getSourceConfig();
+        this.offsetStorageReader = sourceConnectorContext.getOffsetStorageReader();
+
+        doInit();
+    }
+
+    private void doInit() {
+        queryPrometheusReq = new QueryPrometheusReq();
+        queryPrometheusReq.setQuery(sourceConfig.getConnectorConfig().getQuery());
+        queryPrometheusReq.setStep(sourceConfig.getConnectorConfig().getStep());
+
+        interval = sourceConfig.getConnectorConfig().getInterval();
+        initTime = sourceConfig.getConnectorConfig().getInitTime();
+
+        url = MessageFormat.format("{0}/{1}", sourceConfig.getConnectorConfig().getAddress(), sourceConfig.getConnectorConfig().getApi());
+
+        httpClient = HttpClientBuilder.create().build();
+    }
+
+    @Override
+    public void start() throws Exception {
+        log.info("prometheus source connector start.");
+
+        Map<String, String> partitionMap = new HashMap<>();
+        partitionMap.put(INSTANCE_ID, sourceConfig.getConnectorConfig().getConnectorId());
+        RecordPartition recordPartition = new RecordPartition(partitionMap);
+        RecordOffset recordOffset = offsetStorageReader.readOffset(recordPartition);
+        if (recordOffset != null) {
+            Long pollOffset = (Long) recordOffset.getOffset().get("queueOffset");
+            if (pollOffset != null) {
+                // use offset
+                startTime = pollOffset;
+            } else if (initTime != null) {
+                // use preset time
+                startTime = initTime;
+            } else {
+                // use real time
+                startTime = System.currentTimeMillis() / 1000;
+            }
+        }
+    }
+
+    @Override
+    public void commit(ConnectRecord record) {
+        startTime += interval;
+    }
+
+    @Override
+    public String name() {
+        return this.sourceConfig.getConnectorConfig().getConnectorName();
+    }
+
+    @Override
+    public void stop() throws Exception {
+        log.info("prometheus source connector stop.");
+    }
+
+    @Override
+    public List<ConnectRecord> poll() {
+        try {
+            queryPrometheusReq.setStart(startTime);
+            queryPrometheusReq.setEnd(startTime + interval);
+
+            HttpPost httpPost = new HttpPost(url);
+            httpPost.setEntity(new StringEntity(JSON.toJSONString(queryPrometheusReq), ContentType.APPLICATION_JSON));
+            CloseableHttpResponse response = httpClient.execute(httpPost);
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
```

Comment:
Is it possible to use a keep-alive connection or reuse http connection here? For example, use a PoolingHttpClientConnectionManager.

Reply:
yes I agree with you. And I found a feature https://github.com/apache/eventmesh/issues/4481 is planned, so I think I can use directly when ready